src/timedate/timedated.c

```diff
@@ -916,6 +916,63 @@ static int method_set_ntp(sd_bus_message *m, void *userdata, sd_bus_error *error
         return sd_bus_reply_method_return(m, NULL);
 }
 
+static int method_list_timezones(sd_bus_message *m, void *userdata, sd_bus_error *error) {
+        sd_bus *bus = sd_bus_message_get_bus(m);
+        Context *c = userdata;
+        int interactive, r;
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        _cleanup_strv_free_ char **zones = NULL;
+        char** zone;
+
+        assert(m);
+        assert(bus);
+        assert(c);
+
+        r = sd_bus_message_read(m, "b", &interactive);
+        if (r < 0)
+                return r;
+
+        r = bus_verify_polkit_async(
+                        m,
+                        CAP_SYS_TIME,
+                        "org.freedesktop.timedate1.list-timezones",
+                        NULL,
+                        interactive,
+                        UID_INVALID,
+                        &c->polkit_registry,
+                        error);
+        if (r < 0)
+                return r;
+        if (r == 0)
+                return 1;
+
+        r = get_timezones(&zones);
+        if (r < 0) {
+                log_error_errno(r, "Failed to read list of time zones: %m");
+                return r;
```

Comment:
The two lines above can be replaced by a single `return log_error_errno(r, "…");`, as all our log functions return the error you passed in again.

That said, I think it would be better to return the error to clients, using sd_bus_error_setf() or so, instead of simply logging about it here.